src/hyperspectral/hyperspectral/band_selection/train.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Azavea
+#
+# Redistribution and use  in source and binary forms,  with or without
+# modification, are  permitted provided that the  following conditions
+# are met:
+#
+# 1. Redistributions  of source code  must retain the  above copyright
+# notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions  and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# 3. Neither  the name of  the copyright holder  nor the names  of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY  THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS"  AND ANY EXPRESS  OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT
+# LIMITED TO,  THE IMPLIED  WARRANTIES OF MERCHANTABILITY  AND FITNESS
+# FOR  A PARTICULAR  PURPOSE ARE  DISCLAIMED.  IN NO  EVENT SHALL  THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT  LIMITED TO,  PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES;
+# LOSS OF  USE, DATA,  OR PROFITS;  OR BUSINESS  INTERRUPTION) HOWEVER
+# CAUSED AND ON  ANY THEORY OF LIABILITY, WHETHER  IN CONTRACT, STRICT
+# LIABILITY, OR  TORT (INCLUDING  NEGLIGENCE OR OTHERWISE)  ARISING IN
+# ANY WAY  OUT OF  THE USE OF  THIS SOFTWARE, EVEN  IF ADVISED  OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import torch as torch
+import torch.nn.functional as F
+
+
+def vanilla_train(model: torch.nn.Module,
```

Comment:
Curious here about the "vanilla" naming?  And while I'm after nits, × (hex code d7) renders more universally.  (I am blushing as I lodge this comment, over it's persnicketyness.  Feel free to disregard.)

Reply:
This is to convey a sense of unadornedness, primitive simplicity, if you will.